### ISSUE TITLE
CA1801 - Remove unused parameter code fix should not attempt to fix custom operators

### ISSUE THREAD

**Issue by bartdesmet** [dotnet/roslyn-analyzers#4697]
For the following fragment:

```csharp
public struct Unit
{
    public static bool operator ==(Unit x, Unit y) => true;
    public static bool operator !=(Unit x, Unit y) => false;
}
```

CA1801 suggests to remove the parameters, resulting in invalid code with

```
CS1534 - Overloaded binary operator takes two parameters.
```

The same holds for unary operators of course. While the warning for CA1801 is good, it's unactionable in terms of a code fix for special members like operators.

**Comment by mavasani** [comment]
Kindly use discard symbol names, such as `_`, `_1`, `_2`, etc. for cases where parameters are intentionally unused: https://github.com/dotnet/roslyn-analyzers/blob/edf95f72a20ed1c090a1639c8ec85bb5854fe09e/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/MicrosoftCodeQualityAnalyzersResources.resx#L1073-L1075

**Comment by bartdesmet** [comment]
Cool; I didn't know this was in already. I had seen the discussion on it and remember that it was also going to be brought up as a language feature to be able to use `_` as a discard for intentionally unused parameters.

**Comment by bartdesmet** [comment]
Should the code fix provider be adjusted not to offer an invalid fix though?

**Comment by mavasani** [comment]
> Should the code fix provider be adjusted not to offer an invalid fix though?

Sounds reasonable.

**Comment by AraHaan** [comment]
How about this have the code fix provide an option to replace the unused param with a discard param esp when you use a trash param to provide an overload due to adding overloads with different params but all taking in the same types of information as the other versions do just to avoid a compile error saying the method was duplicated when in fact it takes in different inputs.

**Comment by mavasani** [comment]
That sounds like a good suggestion!

**Comment by Youssef1313** [comment]
@mavasani, Should *both* fixes (the existing one and the new suggested one) be offered for normal methods (not operator overloads)?

**Comment by mavasani** [comment]
@Youssef1313 It makes sense to offer the old fix (parameter removal) only for non-operator methods and new fix (rename parameter to have a discard symbol name) for all methods.

**Comment by mavasani** [comment]
@Youssef1313 Also note that we likely want to port the code fix to the corresponding IDE rule, as CA1801 will be deprecated starting .NET 6.0 SDK. See https://github.com/dotnet/roslyn-analyzers/issues/4498

**Comment by Youssef1313** [comment]
@bartdesmet Given that this analyzer is being removed in .NET 6.0, I dropped the fix of https://github.com/dotnet/roslyn-analyzers/pull/4700.

**Comment by mavasani** [comment]
I think we can close this issue. https://github.com/dotnet/roslyn/issues/49868#issuecomment-761024176 tracks handling this case when add a code fix for IDE0060
